mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -761,16 +783,24 @@ class GeneralizedLinearRegressionModel private[ml] (
    * otherwise generates a new column and sets it as the prediction column on a new copy
    * of the current model.
    */
-  private[regression] def findSummaryModelAndPredictionCol()
-    : (GeneralizedLinearRegressionModel, String) = {
+  private[regression] def getSummaryModel: GeneralizedLinearRegressionModel = {
     $(predictionCol) match {
       case "" =>
         val predictionColName = "prediction_" + java.util.UUID.randomUUID.toString
-        (copy(ParamMap.empty).setPredictionCol(predictionColName), predictionColName)
-      case p => (this, p)
+        copy(ParamMap.empty).setPredictionCol(predictionColName)
+      case p => this
```

Comment:
I'd like to change it to ```case p => copy(ParamMap.empty)```, and this will make ```GeneralizedLinearRegressionTrainingSummary``` always along with an immutable model.
Users want to learn about the training summary is a specific model rather than an updated model. If model was updated later not by retrain, we should not provide updated training summary.